File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        l= 0 
        r = 0
        maxp =0

        while r<len(prices):
            if prices[l]<prices[r]:
                pr = prices[r]-prices[l]
                maxp = max(pr,maxp)
            else:
                l=r
            r+=1

        return maxp
        



        
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0;
        int r = 0;
        int maxp = 0;

        while (r<prices.length){
            if (prices[l]<prices[r]){
                int diff = prices[r]-prices[l];
                maxp = Math.max(diff,maxp);
            }
            else{
                l = r;
            }
            r++;
        }

        return maxp;

    }
}